tensorflow-framework/src/main/java/org/tensorflow/framework/optimizers/Optimizer.java

```diff
@@ -56,70 +63,68 @@ protected Optimizer(Graph graph) {
     this.tf = Ops.create(graph).withName(getOptimizerName());
     this.slots = new HashMap<>();
     this.globals = new ArrayList<>();
+    setLearningRate(LEARNING_RATE_DEFAULT);
   }
 
   /**
    * Builds an optimizer for the supplied graph.
    *
+   * <p>Uses the name from {@link Optimizer#getOptimizerName()} to name the operations.
+   *
    * @param graph The graph to optimize.
-   * @param name The base name for the operations.
+   * @param learningRate the learning rate.
    */
-  protected Optimizer(Graph graph, String name) {
+  protected Optimizer(Graph graph, float learningRate) {
     this.graph = graph;
-    this.tf = Ops.create(graph).withName(name);
+    this.tf = Ops.create(graph).withName(getOptimizerName());
     this.slots = new HashMap<>();
     this.globals = new ArrayList<>();
+    setLearningRate(learningRate);
   }
 
   /**
-   * Gets the Optimizer's Ops instance
-   * @return the Optimizer's Ops instance
+   * Builds an optimizer for the supplied graph.
+   *
+   * @param graph The graph to optimize.
+   * @param name The base name for the operations.
    */
-  public final Ops getTF() {
-    return tf;
+  protected Optimizer(Graph graph, String name) {
+    this.graph = graph;
+    this.tf = Ops.create(graph).withName(name);
+    this.slots = new HashMap<>();
+    this.globals = new ArrayList<>();
+    setLearningRate(LEARNING_RATE_DEFAULT);
   }
 
   /**
-   * Creates a name by combining a variable name and a slot name
+   * Builds an optimizer for the supplied graph.
    *
-   * @param variable the variable
-   * @param slotName the name of the slot
-   * @return the combined name
+   * @param graph The graph to optimize.
+   * @param name The base name for the operations.
+   * @param learningRate the learning rate.
    */
+  protected Optimizer(Graph graph, String name, float learningRate) {
+    this.graph = graph;
+    this.tf = Ops.create(graph).withName(name);
+    this.slots = new HashMap<>();
+    this.globals = new ArrayList<>();
+    setLearningRate(learningRate);
+  }
+
   public static String createName(Output<? extends TType> variable, String slotName) {
     return variable.op().name() + "-" + slotName;
   }
 
-  /**
```

Comment:
Why'd the Javadoc go away?

Reply:
I am not sure what happened. I had a local copy that I saved and it was there, so will add it back in.

Reply:
Update pushed